synapseclient/core/utils.py

```diff
@@ -1443,49 +1443,3 @@ def merge_dataclass_entities(
         setattr(destination, key, value)
 
     return destination
-
-
-def merge_non_modifiable_manifest_fields(
```

Comment:
Why was this function removed in favor of `_determine_fields_to_ignore_in_merge` added to             `merge_dataclass_entities` in the file's store function?

Reply:
I wrote it in this way to keep the utility class here with the least amount of responsibility. If we determine the fields we are not going to merge before calling this utility function it'll mean that this code can be generically re-used for other Entities. File was the first entity to have drastically different rules determining how data in merged and it made sense to leave it within the context of the File